### PR TITLE
[corenfc] Update for Xcode 11 beta 1

### PR DESCRIPTION
**Pull request by spouliot**
There was no changes in beta 2

There's a generator issue https://github.com/xamarin/xamarin-macios/issues/6366
that prevent a fully correct `NFCNdefTag` protocol

### CODE REVIEW

**Comment by spouliot** [comment]
known issues

* mono-native-//Modern: #6322
* msbuild tests/iOS: #6325

File: src/corenfc.cs [hunk 1]
```diff
@@ -144,9 +150,141 @@ interface NFCIso15693Tag : NFCTag {
...
+#endif
+		[Export ("lockDFSIDWithRequestFlag:completionHandler:")]
+		void LockDfsi (RequestFlag flags, Action<NSError> completionHandler);
+
+		[iOS (13,0)]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("getSystemInfoWithRequestFlag:completionHandler:")]
+		void GetSystemInfo (RequestFlag flags, Action<nint, nint, nint, nint, nint, NSError> completionHandler);
```
Comment (chamons) [inline src/corenfc.cs]: `Action<nint, nint, nint, nint, nint, NSError>` makes my head spin

File: src/corenfc.cs [hunk 2]
```diff
@@ -305,5 +559,376 @@ interface NSUserActivity_CoreNFC {
...
+
+	interface INFCNdefTag {}
+
+	delegate void NFCQueryNdefStatusCompletionHandler (NFCNdefStatus status, nuint capacity, NSError error);
+
+	[iOS (13,0)]
+	[Protocol (Name="NFCNDEFTag")]
+	interface NFCNdefTag : NSSecureCoding, NSCopying {
+
+		// FIXME https://github.com/xamarin/xamarin-macios/issues/6366
```
Comment (chamons) [inline src/corenfc.cs]: Maybe add to https://github.com/xamarin/xamarin-macios/issues/6212?

File: src/corenfc.cs (hunk 1 above)
Comment (spouliot) [reply src/corenfc.cs]: oops, missed that one (so many ugly ones)

File: src/corenfc.cs [hunk 3]
```diff
@@ -305,5 +559,376 @@ interface NSUserActivity_CoreNFC {
...
+	[Protocol]
+	interface NFCFeliCaTag : NFCTag, NFCNdefTag {
+
+		[Abstract]
+		[Export ("currentSystemCode", ArgumentSemantic.Retain)]
+		NSData CurrentSystemCode { get; }
+
+		[Abstract]
+		[Export ("currentIDm", ArgumentSemantic.Retain)]
+		NSData CurrentIDm { get; }
```
[the remark this answers is not in the document]
Comment (spouliot) [reply src/corenfc.cs]: I was not sure about that acronym because of the `m` - but (upon searching) it's not a typo

File: src/corenfc.cs [hunk 4]
```diff
@@ -144,9 +152,141 @@ interface NFCIso15693Tag : NFCTag {
 		[Abstract]
 		[Export ("readMultipleBlocksWithConfiguration:completionHandler:")]
 		void ReadMultipleBlocks (NFCIso15693ReadMultipleBlocksConfiguration readConfiguration, Action<NSData, NSError> completionHandler);
+
+		[iOS (13,0)]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("stayQuietWithCompletionHandler:")]
+		void StayQuiet (Action<NSError> completionHandler);
```
Comment (mandel-macaque) [inline src/corenfc.cs]: Do we need async here since it takes a handler?

File: src/corenfc.cs [hunk 5]
```diff
@@ -144,9 +152,141 @@ interface NFCIso15693Tag : NFCTag {
...
+#endif
+		[Export ("extendedLockBlockWithRequestFlags:blockNumber:completionHandler:")]
+		void ExtendedLockBlock (RequestFlag flags, nint blockNumber, Action<NSError> completionHandler);
+
+		[iOS (13,0)]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("extendedReadMultipleBlocksWithRequestFlags:blockRange:completionHandler:")]
+		void ExtendedReadMultipleBlocks (RequestFlag flags, NSRange blockRange, Action<NSData [], NSError> completionHandler);
```
Comment (mandel-macaque) [inline src/corenfc.cs]: All the above have handler, do we need to use Async?

File: src/corenfc.cs [hunk 6]
```diff
@@ -305,5 +561,376 @@ interface NSUserActivity_CoreNFC {
...
+		[Export ("readNDEFWithCompletionHandler:")]
+		void ReadNdef (Action<NFCNdefMessage, NSError> completionHandler);
+
+		[Abstract]
+		[Export ("writeNDEF:completionHandler:")]
+		void WriteNdef (NFCNdefMessage ndefMessage, Action<NSError> completionHandler);
+
+		[Abstract]
+		[Export ("writeLockWithCompletionHandler:")]
+		void WriteLock (Action<NSError> completionHandler);
```
Comment (mandel-macaque) [inline src/corenfc.cs]: Async?

File: src/corenfc.cs (hunk 4 above)
Comment (spouliot) [reply src/corenfc.cs]: no, it's a protocol

File: src/corenfc.cs (hunk 5 above)
Comment (spouliot) [reply src/corenfc.cs]: nope

File: src/corenfc.cs (hunk 6 above)
Comment (spouliot) [reply src/corenfc.cs]: no :)

**Comment by spouliot** [comment]
known issues

* mono-native-//Modern: #6322
* xammac tests: mono/mono#14975
* msbuild tests/iOS: #6325

File: src/corenfc.cs [hunk 7]
```diff
@@ -163,21 +303,68 @@ interface NFCNdefPayload : NSSecureCoding {
...
+		[Static]
+		[Export ("wellKnownTypeURIPayloadWithURL:")]
+		[return: NullAllowed]
+		NFCNdefPayload CreateWellKnowType (NSUrl url);
+
+		[iOS (13,0)]
+		[Static]
+		[Export ("wellKnowTypeTextPayloadWithString:locale:")]
+		[return: NullAllowed]
+		NFCNdefPayload CreateWellKnowType (string text, NSLocale locale);
```
[the remark this answers is not in the document]
Comment (mandel-macaque) [reply src/corenfc.cs]: Puff and I filed a rdar a long time ago with a typo and they said they would not fix it 🤦‍♂

[the remark this answers is not in the document]
Comment (spouliot) [reply src/corenfc.cs]: heh, pretty sure I copy pasted names from the last one
I'll file it - indivisual team/engineer might do better than others :)

**Pull request merged**